compare.go

```diff
@@ -73,63 +93,57 @@ func platformVector(platform specs.Platform) []specs.Platform {
-				if armMinor > 9 {
-					armMinor = 9
-				}
-				armMajor = 8
-				vector = append(vector, specs.Platform{
-					Architecture: platform.Architecture,
-					OS:           platform.OS,
-					OSVersion:    platform.OSVersion,
-					OSFeatures:   platform.OSFeatures,
-					Variant:      "v8." + strconv.Itoa(armMinor),
-				})
-			}
-
-			for minor := armMinor - 1; minor >= 0; minor-- {
-				arm64Variant := "v" + strconv.Itoa(armMajor) + "." + strconv.Itoa(minor)
-				if minor == 0 {
-					arm64Variant = "v" + strconv.Itoa(armMajor)
-				}
-				vector = append(vector, specs.Platform{
-					Architecture: platform.Architecture,
-					OS:           platform.OS,
-					OSVersion:    platform.OSVersion,
-					OSFeatures:   platform.OSFeatures,
-					Variant:      arm64Variant,
-				})
-			}
+		vector = []specs.Platform{} // Reset vector, `arm64PlatformVector` will add the first variant
+		switch variant {
+		case "v8", "v8.0":
+			vector = append(vector, arm64PlatformVector(platform, 8, 0)...)
+		case "v8.1":
+			vector = append(vector, arm64PlatformVector(platform, 8, 1)...)
+		case "v8.2":
+			vector = append(vector, arm64PlatformVector(platform, 8, 2)...)
+		case "v8.3":
+			vector = append(vector, arm64PlatformVector(platform, 8, 3)...)
+		case "v8.4":
+			vector = append(vector, arm64PlatformVector(platform, 8, 4)...)
+		case "v8.5":
+			vector = append(vector, arm64PlatformVector(platform, 8, 5)...)
+		case "v8.6":
+			vector = append(vector, arm64PlatformVector(platform, 8, 6)...)
+		case "v8.7":
+			vector = append(vector, arm64PlatformVector(platform, 8, 7)...)
+		case "v8.8":
+			vector = append(vector, arm64PlatformVector(platform, 8, 8)...)
+		case "v8.9":
+			vector = append(vector, arm64PlatformVector(platform, 8, 9)...)
+		case "v9", "v9.0":
+			vector = append(vector, arm64PlatformVector(platform, 9, 0)...)
+			vector = append(vector, arm64PlatformVector(platform, 8, 5)...)
+		case "v9.1":
+			vector = append(vector, arm64PlatformVector(platform, 9, 1)...)
+			vector = append(vector, arm64PlatformVector(platform, 8, 6)...)
+		case "v9.2":
+			vector = append(vector, arm64PlatformVector(platform, 9, 2)...)
+			vector = append(vector, arm64PlatformVector(platform, 8, 7)...)
+		case "v9.3":
+			vector = append(vector, arm64PlatformVector(platform, 9, 3)...)
+			vector = append(vector, arm64PlatformVector(platform, 8, 8)...)
+		case "v9.4":
+			vector = append(vector, arm64PlatformVector(platform, 9, 4)...)
+			vector = append(vector, arm64PlatformVector(platform, 8, 9)...)
+		case "v9.5":
+			vector = append(vector, arm64PlatformVector(platform, 9, 5)...)
+			vector = append(vector, arm64PlatformVector(platform, 8, 9)...)
+		case "v9.6":
+			vector = append(vector, arm64PlatformVector(platform, 9, 6)...)
+			vector = append(vector, arm64PlatformVector(platform, 8, 9)...)
+		case "v9.7":
+			vector = append(vector, arm64PlatformVector(platform, 9, 7)...)
+			vector = append(vector, arm64PlatformVector(platform, 8, 9)...)
```

Comment:
I think you can get rid of this entire switch/case section with something like this map: https://go.dev/play/p/_eOLkGoEdbA

The map will return the values you need to call `arm64PlatformVector`; just need a simple loop so that both "8" and "9" variants get added once you get to v9 and above.

Reply:
Updated